spec/ad_parser.spec.js

```diff
@@ -14,24 +20,261 @@ describe('AdParser', () => {
         '<Ad><Wrapper><VASTAdTagURI>foo</VASTAdTagURI></Wrapper></Ad>'
       );
       invalidAdNode = getNodesFromXml('<Ad><Foo></Foo></Ad>');
+      adElement = getNodesFromXml(linearAd);
+    });
+
+    beforeEach(() => {
+      ad = parseAd(adElement, null);
     });
 
     it('correctly returns inline and passes ad attributes down', () => {
       expect(parseAd(inlineAdNode, emit)).toEqual({
-        ad: expect.objectContaining({ id: 'id-123', sequence: 'seq-123' }),
-        type: 'INLINE'
+        adServingId: null,
+        adType: null,
+        adVerifications: [],
+        advertiser: null,
+        categories: [],
+        creatives: [],
+        description: null,
+        errorURLTemplates: [],
+        expires: null,
+        extensions: [],
+        id: 'id-123',
+        impressionURLTemplates: [],
+        pricing: null,
+        sequence: 'seq-123',
+        survey: null,
+        system: null,
+        title: null,
+        viewableImpression: {}
```

Comment:
I'm not sure how this unit test is passing, because `parseAd` returns an object `{ ad: Object, type: 'WRAPPER'|'INLINE' }`